File: payment-service/src/main/java/com/hedvig/paymentservice/domain/trustlyOrder/commands/PayoutErrorReceivedCommand.java
package com.hedvig.paymentservice.domain.trustlyOrder.commands;

import java.util.UUID;

import com.hedvig.paymentService.trustly.data.response.Error;

import org.axonframework.commandhandling.TargetAggregateIdentifier;

import lombok.Value;

@Value
public class PayoutErrorReceivedCommand {
    @TargetAggregateIdentifier
    UUID hedvigOrderId;

    Error error;
}

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/trustlyOrder/commands/PayoutErrorReceivedCommand.kt
package com.hedvig.paymentservice.domain.trustlyOrder.commands

import com.hedvig.paymentService.trustly.data.response.Error
import java.util.*
import org.axonframework.commandhandling.TargetAggregateIdentifier

data class PayoutErrorReceivedCommand(
    @TargetAggregateIdentifier  
    val hedvigOrderId: UUID,
    val error: Error
)
